docs/architecture/adr-006-support-multisig-chain-link.md

```diff
@@ -0,0 +1,320 @@
+# ADR 006: Support multisig chain link
+
+## Changelog
+
+- September 29th 2021: Initial draft
+- October 21th, 2021: Proposed
+- October 21th, 2021: First Review
+- October 22th, 2021: Second Review
+
+## Status
+
+PROPOSED
+
+## Abstract
+
+Currently, it is not possible to create a chain link using a multisig account. Since many validators MIGHT use multisig accounts, we SHOULD change the `Proof` type and its `Verify` function in order to support this kind of account as well.
+
+## Context
+
+Currently, the `x/profiles` module gives users the possibility to link their profile to different external accounts. 
+In particular, to link other blockchains accounts to a profile, we follow this process:
+1. the user signs a message with their own private key;
+2. the signature and the signed value are placed inside a `Proof` object;
+3. Desmos verifies the `Proof` object to guarantee that the user really owns such account and thus it can be linked successfully to their Desmos profile.
+
+Currently, this process works properly for single-signature accounts, but it does not support multi-signature accounts. This is due to the fact that the `Proof` type only supports signatures made by a single-signature account, and its `Verify` function is only able to verify such signature type.
+
+## Decision
+
+We propose to change the `Signature` of `Proof` into be 
+[SignatureDescriptor_Data](https://github.com/cosmos/cosmos-sdk/blob/master/proto/cosmos/tx/signing/v1beta1/signing.proto#L57) instance.
+
+### Proof implementation
+
+`SignatureDescriptor_Data` supports to store both single and multi signatures. Moreover, it has a 
+[function](https://github.com/cosmos/cosmos-sdk/blob/master/types/tx/signing/signature.go#L65) 
+to convert to [SignatureData](https://github.com/cosmos/cosmos-sdk/blob/master/types/tx/signing/signature_data.go#L10), 
+which is helpful to the signature verification in `Verify` function. The verification process will be like:
+1. If it's a `SingleSignatureData`, make sure the account public key is a cryptotypes. `PubKey` and 
+then use the `VerifySignature` method to verify the signature.
+2. If it's a `MultiSignatureData`, make sure the account public key is a `multisig.PubKey` and 
+then use the `VerifyMultisignature` method to verify the signature.
+
+The whole process in code will be like:
```

Comment:
I think the code below can be safely deleted since we are already describing above how things are implemented anyway